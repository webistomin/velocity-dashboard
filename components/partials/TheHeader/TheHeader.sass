+b('page-header')
  grid-area: header

  +e('container')
    position: relative
    display: flex
    align-items: center
    justify-content: space-between
    border-bottom: 1px solid var(--color-border-gray)

    +media($screen-sm)
      height: rem(80px)
      box-sizing: border-box
      padding-left: rem(9px)
      padding-right: rem(17px)

    +media($screen-lg)
      padding-left: rem(112px)
      padding-right: rem(105px)

  +e('logo')
    display: none

    +media($screen-sm)
      display: block
      +absolute-center()

    +media($screen-lg)
      position: static
      top: auto
      left: auto
      transform: initial

  +e('user-list')
    display: flex

  +e('user-item')
    +m('user')
      display: none

      +media($screen-xs)
        display: block

    +m('messages')
      display: none

      +media($screen-xs)
        display: block

  +e('nav-toggle')
    +media($screen-lg)
      display: none

  +e('title')
    +absolute-center()
    +text-overflow()
    font: var(--text-heading-4)
    max-width: calc(100% - 10rem)

    +media($screen-xs)
      max-width: calc(100% - 20rem)

    +media($screen-sm)
      display: none
