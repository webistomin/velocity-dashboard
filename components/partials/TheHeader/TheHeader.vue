<template lang="pug">
  header.page-header
    BaseLogo
</template>

<script lang="ts">
  import Vue from 'vue';
  import BaseLogo from '~/components/base/BaseLogo';

  export default Vue.extend({
    name: 'TheHeader',
    components: { BaseLogo },
  });
</script>

<style lang="sass">
  @import "TheHeader"
</style>
