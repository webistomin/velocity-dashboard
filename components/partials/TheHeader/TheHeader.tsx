import { VueComponent } from 'types/vue-components';
import { Component } from 'nuxt-property-decorator';
import { VNode } from 'vue';

import BaseLogo from 'components/base/BaseLogo';
import NotificationCenter from 'components/ui/NotificationCenter';
import MessagesCenter from 'components/ui/MessagesCenter';
import UserCenter from 'components/ui/UserCenter';
import BaseToggle from 'components/base/BaseToggle';

import './TheHeader.sass';

@Component({
  name: 'TheHeader',
})
export default class TheHeader extends VueComponent {
  public render(): VNode {
    return (
      <header class='page-header'>
        <div class='page-header__container'>
          <BaseToggle class='page-header__nav-toggle' />
          <BaseLogo class='page-header__logo' />
          <h2 class='page-header__title title'>Page title</h2>
          <ul class='page-header__user-list list'>
            <li class='page-header__user-item page-header__user-item_notification list-item'>
              <NotificationCenter />
            </li>
            <li class='page-header__user-item page-header__user-item_messages list-item'>
              <MessagesCenter />
            </li>
            <li class='page-header__user-item page-header__user-item_user list-item'>
              <UserCenter />
            </li>
          </ul>
        </div>
      </header>
    );
  }
}
