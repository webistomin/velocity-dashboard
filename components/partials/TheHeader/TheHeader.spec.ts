import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import TheHeader from './TheHeader';

describe('TheHeader', () => {
  const toggleSelector = '.page-header__nav-toggle';
  const localVue = createLocalVue();
  localVue.use(VueRouter);
  const router = new VueRouter();

  const factory = (propsData = {}, options = {}) => {
    return shallowMount(TheHeader, {
      localVue,
      router,
      propsData: {
        isNavOpened: false,
        ...propsData,
      },
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });

  it('Emit "openNav" event from button', () => {
    const wrapper = factory();
    const btn = wrapper.find(toggleSelector);
    btn.trigger('click');
    expect(wrapper.emitted().openNav);
  });

  it('Emit "openNav" event', () => {
    const wrapper = factory();
    wrapper.vm.onToggleClick();
    expect(wrapper.emitted().openNav);
  });
});
