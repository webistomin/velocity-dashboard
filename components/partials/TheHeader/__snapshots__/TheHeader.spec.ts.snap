// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheHeader Matches snapshot 1`] = `
"<header class=\\"page-header\\">
  <div class=\\"page-header__container container\\">
    <basetoggle-stub size=\\"xs\\" class=\\"page-header__nav-toggle\\"></basetoggle-stub>
    <vuecomponent-stub class=\\"page-header__logo\\"></vuecomponent-stub>
    <h2 class=\\"page-header__title title\\"></h2>
    <ul class=\\"page-header__user-list list\\">
      <li class=\\"page-header__user-item page-header__user-item_notification list-item\\">
        <notificationcenter-stub></notificationcenter-stub>
      </li>
      <li class=\\"page-header__user-item page-header__user-item_messages list-item\\">
        <messagescenter-stub></messagescenter-stub>
      </li>
      <li class=\\"page-header__user-item page-header__user-item_user list-item\\">
        <usercenter-stub></usercenter-stub>
      </li>
    </ul>
  </div>
</header>"
`;
